music/src/music_vae/midi_me.ts

```diff
@@ -0,0 +1,299 @@
+import * as tf from '@tensorflow/tfjs';
+import * as logging from '../core/logging';
+export {MidiMe};
+
+/**
+ * Class for transforming a single variable Gaussian distribution into a
+ * multivariate one.
+ */
+class SamplingLayer extends tf.layers.Layer {
+  constructor() {
+    super({});
+  }
+
+  computeOutputShape(inputShape: tf.Shape[]) {
+    return inputShape[0];
+  }
+
+  call(inputs: [tf.Tensor2D, tf.Tensor2D]) {
+    return tf.tidy(() => {
+      const [mu, sigma] = inputs;
+      return tf.add(tf.mul(tf.randomNormal(sigma.shape), sigma), mu);
+    });
+  }
+  getClassName() {
+    return 'SamplingLayer';
+  }
+}
+
+/**
+ * An interface for providing configurable properties to the MidiMe model.
+ * @param input_size The shape of the VAE input. Since the inputs to this
+ * VAE are actually latent vectors from MusicVAE, then this number should be
+ * equal to the number of latent variables used by MusicVAE (`zDims`). The
+ * default is 256.
+ * @param output_size The size of the output. The default is 4.
```

Comment:
By output do you mean the number of dimensions of the MidiMe latent vector? Please use a clearer name and description.